Modules/_zoneinfo.c

```diff
@@ -2243,72 +2246,123 @@ ymd_to_ord(int y, int m, int d)
+/* Convert a generic PyObject into a PyDateTime_DateTime object. */
+static PyDateTime_DateTime *
+as_datetime(PyObject *dt)
 {
-    assert(local_ts != NULL);
-
+    int year, month, day;
     int hour, minute, second;
-    int ord;
+    unsigned char fold;
     if (PyDateTime_CheckExact(dt)) {
-        int y = PyDateTime_GET_YEAR(dt);
-        int m = PyDateTime_GET_MONTH(dt);
-        int d = PyDateTime_GET_DAY(dt);
+        year = PyDateTime_GET_YEAR(dt);
+        month = PyDateTime_GET_MONTH(dt);
+        day = PyDateTime_GET_DAY(dt);
         hour = PyDateTime_DATE_GET_HOUR(dt);
         minute = PyDateTime_DATE_GET_MINUTE(dt);
         second = PyDateTime_DATE_GET_SECOND(dt);
-
-        ord = ymd_to_ord(y, m, d);
+        fold = PyDateTime_DATE_GET_FOLD(dt);
     }
     else {
-        PyObject *num = PyObject_CallMethod(dt, "toordinal", NULL);
+        PyObject *num = PyObject_CallMethod(dt, "year", NULL);
         if (num == NULL) {
-            return -1;
+            return NULL;
+        }
+        year = PyLong_AsLong(num);
+        Py_DECREF(num);
+        if (year == -1) {
+            return NULL;
         }
 
-        ord = PyLong_AsLong(num);
+        num = PyObject_GetAttrString(dt, "month");
+        if (num == NULL) {
+            return NULL;
+        }
+        month = PyLong_AsLong(num);
         Py_DECREF(num);
-        if (ord == -1 && PyErr_Occurred()) {
-            return -1;
+        if (month == -1) {
+            return NULL;
+        }
+
+        num = PyObject_GetAttrString(dt, "day");
+        if (num == NULL) {
+            return NULL;
+        }
+        day = PyLong_AsLong(num);
+        Py_DECREF(num);
+        if (day == -1) {
+            return NULL;
         }
 
         num = PyObject_GetAttrString(dt, "hour");
         if (num == NULL) {
-            return -1;
+            return NULL;
         }
         hour = PyLong_AsLong(num);
         Py_DECREF(num);
         if (hour == -1) {
-            return -1;
+            return NULL;
         }
 
         num = PyObject_GetAttrString(dt, "minute");
         if (num == NULL) {
-            return -1;
+            return NULL;
         }
         minute = PyLong_AsLong(num);
         Py_DECREF(num);
         if (minute == -1) {
-            return -1;
+            return NULL;
         }
 
         num = PyObject_GetAttrString(dt, "second");
         if (num == NULL) {
-            return -1;
+            return NULL;
         }
         second = PyLong_AsLong(num);
         Py_DECREF(num);
         if (second == -1) {
-            return -1;
+            return NULL;
         }
+
+        num = PyObject_GetAttrString(dt, "fold");
+        if (num == NULL) {
+            return NULL;
+        }
+        fold = PyLong_AsLong(num);
+        Py_DECREF(num);
+        if (fold > 1) {
+            return NULL;
+        }
+    }
+
+    return (PyDateTime_DateTime *) PyDateTime_FromDateAndTimeAndFold(
+        year, month, day, hour, minute, second, 0, fold
```

Comment:
<!-- Questions are appropriate if you have a potential concern but are not quite sure if it's relevant or not. Asking the author for clarification or investigation can lead to a quick resolution.  -->
**question:** ‏This drops any `tzinfo` information. Is that a problem?

Reply:
I share the same concern but there doesn't seem to be a constructor for that option

Reply:
Sorry for tagging but I saw @pganssle you were in the commit history. Perhaps you could kindly share your thoughts here?